drivers/SmartThings/zigbee-switch/src/tuya-multi/init.lua

```diff
@@ -0,0 +1,123 @@
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local data_types = require "st.zigbee.data_types"
+local zcl_messages = require "st.zigbee.zcl"
+local generic_body = require "st.zigbee.generic_body"
+local messages = require "st.zigbee.messages"
+local zb_const = require "st.zigbee.constants"
+local read_attribute = require "st.zigbee.zcl.global_commands.read_attribute"
+
+local DISCOVER_ATTR_RSP_ID = 0x0D
+local DISCOVER_ATTR_RST_ID = 0x0C
+local TUYA_MFR_HEADER = "_TZ"
+
+local function is_tuya_products(opts, driver, device)
+    if string.sub(device:get_manufacturer(),1,3) == TUYA_MFR_HEADER then  -- if it is a tuya device, then send the magic packet
+        local subdriver = require("tuya-multi")
+        return true, subdriver
+    end
+    return false
+end
+
+local function send_discover_command(device)
+    local discover_command_payload = "\x00\x00\xff"      -- Start Attribute: 0x0000, Maximum Number: 255
+    local header_args = {
+      cmd = data_types.ZCLCommandId(DISCOVER_ATTR_RST_ID)
+    }
+    local zclh = zcl_messages.ZclHeader(header_args)
+    local addrh = messages.AddressHeader(
+      zb_const.HUB.ADDR,
+      zb_const.HUB.ENDPOINT,
+      device:get_short_address(),
+      device:get_endpoint(clusters.Basic.ID),
+      zb_const.HA_PROFILE_ID,
+      clusters.Basic.ID
+    )
+    local message_body = zcl_messages.ZclMessageBody({
+      zcl_header = zclh,
+      zcl_body = generic_body.GenericBody(discover_command_payload)
+    })
+    local send_message = messages.ZigbeeMessageTx({
+        address_header = addrh,
+        body = message_body
+    })
+    device:send(send_message)
+end
+
+local function read_attribute_function(device, cluster_id, attr_id)
```

Comment:
As mentioned, this code is already defined in the zigbee cluster_base, so you can just require the BasicCluster and then use `BasicCluster.read_attribute(device, cluster_id, attr_id)` instead of duplicating the code here.